transcom/mymove
# INT B-19553 sit dashboard and prime sim updates

## [Agility ticket](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-19553)

## Summary

On the MTO page there is a SIT dashboard that pulls all of its data from an object that is calculated (not stored) in the backend. This object is called `sitStatus` and is configured in `pkg/services/sit_status/shipment_sit_status.go` and returned and used in the UI.

The problem was that, when using the prime simulator, TOOs & TIOs were not seeing these values updating in the SIT dashboard for destination SIT service items:
- `sitDepartureDate`
- `sitRequestedDelivery`
- `sitCustomerContacted`
It was found out that these were supposed to be coming from `DDASIT` service items in each shipment, but for some reason it was never populating and was always `nil`. Even when directly updating that service item in the database.

But why tho?
There was a sneaky private function that was running and returning the wrong service item values and essentially updating the struct with `nil` values. In this case, `DOASIT` service item values were overwriting `DDASIT` service item values because the backend was not checking for what the current SIT was - just always returning `DOASIT` values.

This PR does the following:
- adds edit capabilities in the prime simulator and Prime API to allow for editing of `DDASIT` service items
- updated swagger and validations to allow for updating `sitDepartureDate`, `sitRequestedDelivery`, and `sitCustomerContacted` values for `DDASIT`
- renamed variables to be more appropriate
- passing in `location` variable to allow for the right service item being returned (`DDASIT` if `location == "DESTINATION"`)
- updated API docs

## Verification Steps for the Author

### How to test

1. Fire up your `client` and `server` and sign in as a multi-role office user
2. Either find a shipment with SIT service items or request SIT as the prime
3. Click into the `Move Task Order` tab after entering that move
4. You'll see the SIT panel - the values we care about are `SIT departure date`, `Customer contact date`, and `Requested delivery date` - these are the one we are going to update in the prime sim
**NOTE** You'll need to make sure that the shipment doesn't have origin SIT or that the current SIT is destination. You want to see `Current location: destination SIT` in the SIT dashboard
5. Change your role to the prime simulator and find that move
6. Find the `Edit` button next to the respective `DDASIT` service item you're wanting to update
7. Update all values, make sure that `SIT departure date` occurs after today's date
8. Go back as the TOO role and find the same SIT dashboard
9. You should now see those values reflected in the panel


## Screenshots

The values we want to change
![Screenshot 2024-05-13 at 3 50 59 PM](https://github.com/transcom/mymove/assets/136510600/3bb51945-5a01-4870-9eb9-30afb0f79a5a)

Edit button added in prime sim for `DDASIT` service items
![Screenshot 2024-05-13 at 4 00 59 PM](https://github.com/transcom/mymove/assets/136510600/d0a33717-bac5-43a3-acfe-3d9000bb2cb4)

Updating values of `DDASIT` service item
![Screenshot 2024-05-13 at 4 02 10 PM](https://github.com/transcom/mymove/assets/136510600/849f4f9b-8686-4d6e-b061-aa420d596449)

Now we see this updated values reflect in the SIT dashboard
![Screenshot 2024-05-13 at 4 02 48 PM](https://github.com/transcom/mymove/assets/136510600/1de47c02-2500-4dfd-b4d6-fd1d902216cc)
